src/ManagedCodecs/TargaCodec.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+
+using SixLabors.ImageSharp;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.PixelFormats;
+
+using PhotoSauce.MagicScaler;
+
+namespace PhotoSauce.ManagedCodecs
+{
+	/// <summary>Represents a TGA image file.</summary>
+	public sealed class TargaContainer : IImageContainer, IDisposable
+	{
+		private readonly Image decodedImage;
+
+		private TargaContainer(Image image) => decodedImage = image;
+
+		/// <inheritdoc />
+		public FileFormat ContainerFormat => FileFormat.Bmp;
+
+		/// <inheritdoc />
+		public int FrameCount => 1;
+
+		/// <inheritdoc />
+		public IImageFrame GetFrame(int index) => index == 0 ? new TargaFrame(decodedImage) : throw new IndexOutOfRangeException("Invalid frame index");
+
+		/// <summary>Loads a TGA image from the specified path.</summary>
+		/// <param name="imgPath">The path to the image file.</param>
+		/// <returns>A <see cref="TargaContainer"/> encapsulating the image.</returns>
+		public static TargaContainer Load(string imgPath)
+		{
+			int bpp = 0;
+			using (var fs = File.OpenRead(imgPath))
+			{
+				var info = Image.Identify(fs, out var fmt);
+				if (fmt.Name != "TGA")
+					throw new InvalidDataException("Not a TGA image");
+
+				bpp = info.PixelType.BitsPerPixel;
+			}
+
+			return bpp switch
+			{
+				 8 => new TargaContainer(Image.Load<L8>(imgPath)),
```

Comment:
just for you info: it can also be 16 or even only 15 bpp, you may want to load it with the pixel type `Bgra5551` then.

Reply:
Thanks.  I saw that, but I don't do anything with the 5:5:5:1 format natively, so I'd end up passing it to WIC to convert for me for now.  I actually quite like the way ImageSharp lets me decode into a format I know how to handle.  Would be nice, though, if there was a cleaner way to say "keep it in the format it's stored in" when decoding without having to have knowledge of the codec implementation.